### PR TITLE
Kubernetes 1.9.0

### PR DESCRIPTION
**Pull request by chuckha**
Feel free to update reviewers. Not sure who else can review this though.

Here is the scanner run: https://scanner.heptio.com/d5e73647443a8712a73d62cde9ef6cab/diagnostics/

Signed-off-by: Chuck Ha <chuck@heptio.com>

### CODE REVIEW

File: packer/payload/prepare-ami.sh [hunk 1]
```diff
@@ -37,24 +44,25 @@ apt-get install -qy \
...
-  curl -sf -O "https://storage.googleapis.com/kubernetes-release/network-plugins/cni-amd64-${CNI_RELEASE}.tar.gz"
+  curl -sf -O "https://storage.googleapis.com/kubernetes-release/network-plugins/${CNI}"
 
   install -o root -g root -m 0755 ./kubeadm /usr/bin/kubeadm
   install -o root -g root -m 0755 ./kubectl /usr/bin/kubectl
   install -o root -g root -m 0755 ./kubelet /usr/bin/kubelet
 
   # Also install CNI
-  mkdir -p /opt/cni
+  mkdir -p /opt/cni/bin
```
Comment (chuckha) [inline packer/payload/prepare-ami.sh]: The new cni tarball does not pack things into `bin/` like the last one used to.

File: packer/payload/systemd/10-kubeadm.conf [hunk 2]
```diff
@@ -4,7 +4,10 @@ Environment="KUBELET_SYSTEM_PODS_ARGS=--pod-manifest-path=/etc/kubernetes/manife
 Environment="KUBELET_NETWORK_ARGS=--network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin"
 Environment="KUBELET_DNS_ARGS=--cluster-dns=10.96.0.10 --cluster-domain=cluster.local"
 Environment="KUBELET_AUTHZ_ARGS=--authorization-mode=Webhook --client-ca-file=/etc/kubernetes/pki/ca.crt"
+# Value should match Docker daemon settings.
```
Comment (chuckha) [inline packer/payload/systemd/10-kubeadm.conf]: This is directly taken from the rpm so we are in sync with upstream kubeadm.

File: templates/kubernetes-cluster.template [hunk 3]
```diff
@@ -308,33 +308,35 @@ Mappings:
   # http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html
   RegionMap:
     us-west-1:
-      '64': ami-39023959
+      '64': ami-ff48489f
     ap-south-1:
-      '64': ami-5399d73c
+      '64': ami-275b0f48
+    eu-west-3:
```
Comment (chuckha) [inline templates/kubernetes-cluster.template]: apparently there is a new region

File: templates/kubernetes-cluster.template [hunk 4]
```diff
@@ -667,8 +669,9 @@ Resources:
               command: "systemctl enable awslogs.service && systemctl start awslogs.service"
             # This joins the existing cluster with kubeadm.  Kubeadm is preinstalled in the AMI this template uses, so
             # all that's left is to join the cluster with the correct token.
+            # TODO(chuckha) stop using --discovery-token-unsafe-skip-ca-verification
             "03-k8s-setup-node":
-              command: !Sub "kubeadm reset && kubeadm join --node-name=\"$(hostname -f)\" --token=${KubeadmToken.Token} ${K8sMasterInstance.PrivateIp}:6443"
+              command: !Sub "kubeadm reset && kubeadm join --node-name=\"$(hostname -f)\" --token=${KubeadmToken.Token} --discovery-token-unsafe-skip-ca-verification ${K8sMasterInstance.PrivateIp}:6443"
```
Comment (chuckha) [inline templates/kubernetes-cluster.template]: @mattmoyer said this is ok for now and gave some suggestions on how to improve this. I'd like to fix this by release time.

https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-join/#discovering-what-cluster-ca-to-trust

URL discovery is the method we will likely use, but I'll explore other options that are less of a deviation from what we're doing already.

File: ci/Dockerfile [hunk 5]
```diff
@@ -5,7 +5,7 @@ RUN apk update && apk add --no-cache curl
 # curl kubectl instead of ADD'ing it, since we want to cache the layer
 RUN curl -sfL \
       -o /usr/local/bin/kubectl \
-      https://storage.googleapis.com/kubernetes-release/release/v1.8.2/bin/linux/amd64/kubectl && \
+      https://storage.googleapis.com/kubernetes-release/release/v1.9.0/bin/linux/amd64/kubectl && \
```
Comment (timothysc) [inline ci/Dockerfile]: We can make this more generic to pull the latest, see : https://github.com/heptio/kube-conformance/blob/master/Makefile#L22

File: packer/payload/systemd/10-kubeadm.conf [hunk 6]
```diff
@@ -4,7 +4,10 @@ Environment="KUBELET_SYSTEM_PODS_ARGS=--pod-manifest-path=/etc/kubernetes/manife
...
+# Defaults are "cgroupfs" for Debian/Ubuntu/OpenSUSE and "systemd" for Fedora/CentOS/RHEL
+Environment="KUBELET_CGROUP_ARGS=--cgroup-driver=cgroupfs"
```
Comment (timothysc) [inline packer/payload/systemd/10-kubeadm.conf]: This should be in the .deb and .rpm package.

File: packer/payload/prepare-ami.sh (hunk 1 above)
Comment (timothysc) [reply packer/payload/prepare-ami.sh]: If we're going to pay-down debt, I think we should use apt-get/yum/or @craigtracey work.

**Comment by timothysc** [comment]
@chuckha refresh to 1.9.1 then feel free to merge, but we need to address the other issues too.

**Pull request closed without merging**